crates/ide/src/ide/assists/rewrite_string.rs

```diff
@@ -0,0 +1,352 @@
+//! Rewrite between different types of strings
+//! - URI literals -> Double quoted strings
+//! - Attribute names <-> Double quoted strings
+//! - Double quoted strings <-> Indented strings
+use std::convert::Infallible;
+
+use super::{AssistKind, AssistsCtx};
+use crate::TextEdit;
+use syntax::ast::{self, AstNode};
+use syntax::semantic::{
+    is_valid_ident, strip_indent, unescape_string, unescape_string_escape, unescape_string_literal,
+    StrippedStringPart, UnescapedStringPart,
+};
+use syntax::SyntaxKind;
+
+/// URI literal -> Double quoted strings
+/// ```nix
+/// https://nixos.org
+/// ```
+/// =>
+/// ```nix
+/// "https://nixos.org"
+/// ```
+pub(super) fn rewrite_uri_to_string(ctx: &mut AssistsCtx<'_>) -> Option<()> {
+    let node = ctx.covering_node::<ast::Literal>()?;
+    // Node should be a URI literal
+    if node.kind()? != ast::LiteralKind::Uri {
+        return None;
+    }
+
+    let token = node.token()?;
+    ctx.add(
+        "rewrite_uri_to_string",
+        "Rewrite the URI literal to a double quoted string",
+        AssistKind::QuickFix,
+        vec![TextEdit {
+            delete: token.text_range(),
+            insert: format!(r#""{}""#, token.text()).into(),
+        }],
+    );
+
+    Some(())
+}
+
+/// Attribute names -> Double quoted strings
+/// ```nix
+/// { foo = bar; }
+/// ```
+/// =>
+/// ```nix
+/// { "foo" = bar; }
+/// ```
+pub(super) fn quote_attr(ctx: &mut AssistsCtx<'_>) -> Option<()> {
+    let node = ctx.covering_node::<ast::Name>()?;
+    let token = node.token()?;
+    let syntax = node.syntax();
+
+    // Name should be an attribute, not a e.g. lambda parameter.
+    if syntax.parent().map_or(true, |parent| {
+        !matches!(parent.kind(), SyntaxKind::ATTR_PATH | SyntaxKind::INHERIT)
+    }) {
+        return None;
+    }
+
+    ctx.add(
+        "quote_attr",
+        "Put the attribute name in double quotes",
+        AssistKind::RefactorRewrite,
+        vec![TextEdit {
+            delete: node.syntax().text_range(),
+            insert: format!(r#""{}""#, token.text()).into(),
+        }],
+    );
+    Some(())
+}
+
+/// Double quoted attributes name -> Attribute names without quotes
+/// ```nix
+/// { "foo" = bar; }
+/// ```
+/// =>
+/// ```nix
+/// { foo = bar; }
+/// ```
+///
+/// Double quoted strings -> Indented strings
+/// ```nix
+/// "foo\nbar\n"
+/// ```
+/// =>
+/// ```nix
+/// ''
+///   foo
+///   bar
+/// ''
+/// ```
+pub(super) fn rewrite_string(ctx: &mut AssistsCtx<'_>) -> Option<()> {
```

Comment:
Please split these two. `String` as `Attr` and `String` as `Expr` are semantically distinct. 

There's a plan to add an `Attr` node wrapper. So it's reasonable to put `quote_attr` and `unquote_attr` together, but not `unquote_attr` and `rewrite_string_to_indented`.